tests/framework/integration/installation_and_upgrade/run_tests

```diff
@@ -149,151 +120,72 @@ wait_for_nodes "$CLIENT_1" "rpm -q kmod-lustre-client || exit 0
+export IML_4_INSTALLED=yup
 ./tests/integration/run_tests -f -c /root/cluster_cfg.json -x ~/test_report_pre_upgrade.xml $TESTS"
 
 echo "Now upgrade the OS and IML..."
 
 # first have to shut down the filesystem so that we don't have incompatible OS/IML issues
 ssh root@$TEST_RUNNER "exec 2>&1; set -xe
 cd /usr/share/chroma-manager/
-unset http_proxy; unset https_proxy
-export UPGRADE_FROM_3=$UPGRADE_FROM_3
 ./tests/integration/run_tests -f -c /root/cluster_cfg.json -x ~/test_report_post_filesystem_stop.xml $TESTS_DIR/test_update_with_yum.py:TestYumUpdate.test_stop_before_update"
 
+# re-enable needed repos needed for the upgraded version
+pdsh -l root -R ssh -S -w $(spacelist_to_commalist $CHROMA_MANAGER ${STORAGE_APPLIANCES[@]} ${WORKERS[@]}) "exec 2>&1; set -xe
+rm -rf /etc/yum.repos.d/vault.centos.org_centos_7.5.1804_extras_x86_64_.repo
+yum-config-manager --enable extras" | dshbak -c
+if [ ${PIPESTATUS[0]} != 0 ]; then
+    exit 1
+fi
+
 . $CHROMA_DIR/tests/framework/integration/utils/upgrade_os.sh
-if ! upgrade_os $TEST_DISTRO_NAME $UPGRADE_DISTRO_VERSION $(spacelist_to_commalist $CHROMA_MANAGER ${STORAGE_APPLIANCES[@]} ${WORKERS[@]}); then
+if ! upgrade_os $TEST_DISTRO_NAME $UPGRADE_DISTRO_VERSION $(spacelist_to_commalist ${STORAGE_APPLIANCES[@]} ${WORKERS[@]}); then
     echo "Upgrade failed"
     exit 1
 fi
 
 # re-enable needed repos needed for the upgraded version
 pdsh -l root -R ssh -S -w $(spacelist_to_commalist $CHROMA_MANAGER ${STORAGE_APPLIANCES[@]} ${WORKERS[@]}) "exec 2>&1; set -xe
-yum-config-manager --enable addon-epel\$(rpm --eval %rhel)-x86_64
-yum-config-manager --enable $COPR_OWNER-$COPR_PROJECT
-yum-config-manager --enable mirror.centos.org_centos_7_extras_x86_64_" | dshbak -c
+yum-config-manager --enable $COPR_OWNER-$COPR_PROJECT"
 if [ ${PIPESTATUS[0]} != 0 ]; then
     exit 1
 fi
 
-if [ -f ~/base.repo.in ]; then
-    STORAGE_SERVER_REPO=~/base.repo.in
-fi
-
-if ! ${UPGRADE_FROM_3:-false}; then
-    # need to get the additional storage server repos on the agent
-    # nodes *BEFORE* the upgrade so that when the upgrade completes
-    # and the sessions to the agent are re-established, the agent
-    # already has access to the needed additional repos
-    # this also simulates the state of the MFL repo when the upgrade
-    # will be done in production
-    pdsh -l root -R ssh -S -w "$(spacelist_to_commalist "${STORAGE_APPLIANCES[@]}")" \
-        "exec 2>&1; set -xe
-    # add any repos required by this test run
-    if [ -n \"$STORAGE_SERVER_REPOS\" ]; then
-        for repo in $STORAGE_SERVER_REPOS; do
-            curl \"\$repo\" >> /etc/yum.repos.d/Intel-Lustre-Agent.repo
-        done
-    fi" | dshbak -c
-    if [ "${PIPESTATUS[0]}" != "0" ]; then
-        exit 1
-    fi
-fi
+# Install and setup new manager
+scp "$CHROMA_DIR"/chroma_support.repo "$CHROMA_DIR"/_topdir/RPMS/noarch/python2-iml-manager{,-cli,-libs}-5* "$CHROMA_DIR"/tests/utils/upgrade.exp root@"$CHROMA_MANAGER":/tmp
```

Comment:
Should the `5` here be an environment variable?

Reply:
maybe at some point. It's not being used elsewhere from what I see.